src/storage/leveldb/store.ts

```diff
@@ -0,0 +1,501 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { IAddressInfo, IAddressMetadata, IBalance, IHistoryTx, IStore, ITokenData, ITokenMetadata, IUtxo, IUtxoFilterOptions, IWalletAccessData, IWalletData } from '../../types';
+import { HDPublicKey } from 'bitcore-lib'
+import path from 'path';
+import LevelAddressIndex from './address_index';
+import LevelHistoryIndex from './history_index';
+import LevelUtxoIndex from './utxo_index';
+import LevelWalletIndex from './wallet_index';
+import LevelTokenIndex from './token_index';
+import walletApi from '../../api/wallet';
+import transaction from '../../utils/transaction';
+import fs from 'fs';
+import { HATHOR_TOKEN_CONFIG } from '../../constants';
+
+export default class LevelDBStore implements IStore {
+  addressIndex: LevelAddressIndex;
+  historyIndex: LevelHistoryIndex;
+  utxoIndex: LevelUtxoIndex;
+  walletIndex: LevelWalletIndex;
+  tokenIndex: LevelTokenIndex;
+  dbpath: string;
+  xpubkey: string;
+
+  constructor(dbroot: string, xpubkey: string) {
+    // The xpubkey in the account or change path?
+    this.xpubkey = xpubkey;
+    const xpub = HDPublicKey.fromString(xpubkey);
+    const dbpath = path.join(dbroot, xpub.publicKey.toString());
+    this.addressIndex = new LevelAddressIndex(dbpath);
+    this.historyIndex = new LevelHistoryIndex(dbpath);
+    this.utxoIndex = new LevelUtxoIndex(dbpath);
+    this.walletIndex = new LevelWalletIndex(dbpath);
+    this.tokenIndex = new LevelTokenIndex(dbpath);
+
+    this.dbpath = dbpath;
+  }
+
+  async close(): Promise<void> {
+    await this.addressIndex.close();
+    await this.historyIndex.close();
+    await this.utxoIndex.close();
+    await this.walletIndex.close();
+    await this.tokenIndex.close();
+  }
+
+  async destroy(): Promise<void> {
+    await this.close();
+    if (process.version < 'v14.14.0') {
+      // rmSync was introduced in v14.14.0, earlier versions should use rmdirSync
+      // https://nodejs.org/docs/latest-v14.x/api/fs.html#fs_fs_rmdirsync_path_options
+      fs.rmdirSync(this.dbpath, { recursive: true });
+    } else {
+      fs.rmSync(this.dbpath, { recursive: true, force: true });
+    }
```

Comment:
This method already returns a promise, why do we need to use the sync methods?

This will only cause clients using our lib and calling destroy() to be slower as it will block the event loop

Reply:
Started using async version or `rm`